### PR TITLE
[Improvement](materialized-view) set job failed when toAgentTaskRequest meet error

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
1. set job failed when toAgentTaskRequest meet error
2. enable expr rewrite on create mv
3. fix alias agg query meet error
4. enable aggstate cast on nereids
5. disable curdate on mv
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/AggStateCastPushDownRule.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import org.apache.doris.nereids.types.DataType;
+
+import com.google.common.base.Preconditions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Cast(AggState(tinyint), AggState(int)) to AggState(cast(tinyint,int))
+ */
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/AggStateCastPushDownRule.java:36]: i don't think this is a right way to resolve aggState type cast problem. AggState type should support cast from A to B by itself in BE, should not rely on planner to change its input types.

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**